src/ops.rs

```diff
@@ -173,6 +176,29 @@ pub fn dispatch(
   (base.sync(), boxed_op)
 }
 
+fn op_is_tty(
+  _state: &Arc<IsolateState>,
+  base: &msg::Base<'_>,
+  _data: libdeno::deno_buf,
+) -> Box<Op> {
+  let builder = &mut FlatBufferBuilder::new();
+  let inner = msg::IsTTYRes::create(
+    builder,
+    &msg::IsTTYResArgs {
+      is_tty: unsafe { libc::isatty(libc::STDIN_FILENO as i32) } != 0,
```

Comment:
This is only for stdin... here's an example where it fails
```
deno < /etc/passwd
```
Now stdin is not a TTY but it may very well be a TTY still...
I think isTTY should take an rid argument.

Reply:
Also, atty does this without unsafe, see e.g. https://github.com/denoland/deno/blob/77114fbda49382e397095d8214bd76996b0cfb57/src/permissions.rs#L87

Reply:
@hayd Yes, I think so... unix is very flexible with FDs - however it's probably not portable to windows.
I think just doing those three is good enough - following the API in `atty` https://github.com/softprops/atty
Probably the op should just check all 3 every time.

Reply:
Perhaps it should just do this at boot-time (instead of an op)?

Reply:
that's unnecessary syscalls

Reply:
Fixed